### PR TITLE
Skip exception when deleting cache directories

### PR DESCRIPTION
**Pull request by aschempp**
Sometimes I got exceptions in the install tool because not all cache files were removed. I think we should ignore that, as it's not gonna be a major issue and breaks the application.

### CODE REVIEW

**Comment by leofeyer** [comment]
Are you sure that this does not result in an incomplete cache?

**Comment by aschempp** [comment]
I'm not… but I had no issues just ignoring it, pretty sure the error happens because of invisible files or something similar that are irrelevant…

**Comment by fritzmg** [comment]
`cache:clear` of Symfony first renames the cache folder and _then_ deletes it. Shouldn't `InstallationController::purgeSymfonyCache` do the same?

**Comment by leofeyer** [comment]
It probably should.

**Comment by aschempp** [comment]
Very good idea! Will adjust the PR

**Comment by aschempp** [comment]
PR is updated

File: src/Controller/InstallationController.php [hunk 1]
```diff
@@ -229,7 +229,7 @@ private function login()
      */
     private function purgeSymfonyCache()
     {
-        $fs = new Filesystem();
+        $filesystem = new Filesystem();
```
Comment (leofeyer) [inline src/Controller/InstallationController.php:232]: Just curious: Why is `$filesystem` better than `$fs`?

Comment (aschempp) [reply src/Controller/InstallationController.php:232]: I just copied the code below from Symfony and that's what they named it 🙈

**Comment by fritzmg** [comment]
New method is surely better :). However it does not solve your original problem, does it? An exception might still occur when removing the old cache dir. (At least I frequently experience that even on the console due to operating system shenanigans.)

Though now it's less likely do occur.

**Comment by aschempp** [comment]
You're right. But I guess in that case we should actually show the message?

**Comment by fritzmg** [comment]
Well, you could also ignore it like in your original PR. Because the next time `purgeSymfonyCache` is called, the old cache directory will be tried to delete anyway, before the current cache directory is renamed. (Though what if an error occurs then again?)

File: src/Controller/InstallationController.php [hunk 2]
```diff
@@ -239,8 +239,23 @@ private function purgeSymfonyCache()
...
+
+            if (!is_writable($realCacheDir)) {
+                throw new \RuntimeException(sprintf('Unable to write in the "%s" directory', $realCacheDir));
+            }
+
+            if ($filesystem->exists($oldCacheDir)) {
+                $filesystem->remove($oldCacheDir);
+            }
+
+            $this->container->get('cache_clearer')->clear($realCacheDir);
```
Comment (leofeyer) [inline src/Controller/InstallationController.php]: Did you test this and make sure that there is a `cache_clearer` service?

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @aschempp.

**Comment by fritzmg** [comment]
Btw. shouldn't the same be added to the `manager-bundle`? Currently, it simply deletes the prod folder: https://github.com/contao/manager-bundle/blob/4.4.2/src/Composer/ScriptHandler.php#L60

**Comment by aschempp** [comment]
That's probably a different thing because the `ScriptHandler` does not run within the Symfony application.

**Pull request merged**